addons/hr_timesheet/views/project_views.xml

```diff
@@ -422,5 +422,17 @@
             <field name="domain">[('is_internal_project', '=', False)]</field>
         </record>
 
+        <record id="project_update_view_search_inherit" model="ir.ui.view">
+            <field name="name">project.update.view.search.inherit</field>
+            <field name="model">project.update</field>
+            <field name="inherit_id" ref="project.project_update_view_search"/>
+            <field name="arch" type="xml">
+                <xpath expr="//filter[@name='my_updates']" position='after'>
+                    <filter string="My Team's Update" name="my_team_updates" domain="[('user_id.employee_parent_id.user_id', '=', uid)]"/>
+                    <filter string="My Department's Updates" name="my_department_updates" domain="[('user_id.employee_id.member_of_department', '=', True)]"/>
+                </xpath>
+            </field>
+        </record>
```

Comment:
Since it is a view for the `project.update` and not for the `project.project` or `project.task`. We could create a XML file for this model instead of having views for 3 models in the same XML file. 🙂 
So, I suggest to create a `project_update_views.xml` file and added this override in that file. 🙂